packages/flutter/test/widgets/async_test.dart

```diff
@@ -66,6 +66,19 @@ void main() {
       ));
       expect(find.text('AsyncSnapshot(ConnectionState.waiting, null, null)'), findsOneWidget);
     });
+    testWidgets('gracefully handles transition from null future with initial data without introducing new states', (WidgetTester tester) async {
+      final String initial = 'initial';
+      final GlobalKey key = new GlobalKey();
+      await tester.pumpWidget(new FutureBuilder<String>(
+        key: key, future: null, builder: snapshotText, initialData: initial,
+      ));
+      expect(find.text('AsyncSnapshot(ConnectionState.none, $initial, null)'), findsOneWidget);
+      final Completer<String> completer = new Completer<String>();
+      await tester.pumpWidget(new FutureBuilder<String>(
+        key: key, future: completer.future, builder: snapshotText, initialData: 'No change',
```

Comment:
Maybe `'Ignored'` instead of `'No change'`. Similarly in other test cases below.